orchagent/orch.cpp

```diff
@@ -185,6 +185,30 @@ void Consumer::drain()
         m_orch->doTask(*this);
 }
 
+string Consumer::dumpTuple(KeyOpFieldsValuesTuple &tuple)
+{
+    string s = getTableName() + getConsumerTable()->getTableNameSeparator() + kfvKey(tuple)
+               + "|" + kfvOp(tuple);
+    for (auto i = kfvFieldsValues(tuple).begin(); i != kfvFieldsValues(tuple).end(); i++)
+    {
+        s += "|" + fvField(*i) + ":" + fvValue(*i);
+    }
+
+    return s;
+}
+
+void Consumer::dumpToSyncTasks(vector<string> &ts)
```

Comment:
>dumpToSyncTasks [](start = 15, length = 15)

dumpTuple() and dumpToSyncTasks() are not well designed interfaces.

I agree the observability is important.

Reply:
dumpTuple() is existing function name.  It is refactored so the common function may be used in multiple places. Any specific issue/suggestion?

Reply:
I will investigate and hopefully submit a PR

---
In reply to: [209787965](https://github.com/Azure/sonic-swss/pull/554#discussion_r209787965) [](ancestors = 209787965)